extension/encoding/awslogsencodingextension/internal/unmarshaler/cloudtraillogs/cloudtraillogs.go

```diff
@@ -0,0 +1,290 @@
+		if err != nil {
+			return plog.Logs{}, fmt.Errorf("failed to create gzip reader: %w", err)
+		}
+	} else {
+		if err := gzipReader.Reset(bytes.NewReader(buf)); err != nil {
+			u.gzipPool.Put(gzipReader)
+			return plog.Logs{}, fmt.Errorf("failed to reset gzip reader: %w", err)
+		}
+	}
+
+	defer func() {
+		_ = gzipReader.Close()
+		u.gzipPool.Put(gzipReader)
+	}()
+
+	decompressedBuf, err := io.ReadAll(gzipReader)
+	if err != nil {
+		return plog.Logs{}, fmt.Errorf("failed to decompress CloudTrail logs: %w", err)
+	}
+
+	var cloudTrailLogs CloudTrailLogs
+	if err := json.Unmarshal(decompressedBuf, &cloudTrailLogs); err != nil || len(cloudTrailLogs.Records) == 0 {
+		return plog.Logs{}, fmt.Errorf("failed to unmarshal CloudTrail logs: %w", err)
+	}
+
+	return u.processRecords(cloudTrailLogs.Records), nil
+}
+
+func (u *CloudTrailLogsUnmarshaler) processRecords(records []CloudTrailRecord) plog.Logs {
+	logs := plog.NewLogs()
+
+	if len(records) == 0 {
+		return logs
+	}
+
+	// Create a single resource logs entry for all records
+	resourceLogs := logs.ResourceLogs().AppendEmpty()
+	scopeLogs := resourceLogs.ScopeLogs().AppendEmpty()
+	scopeLogs.Scope().SetName(metadata.ScopeName)
+	scopeLogs.Scope().SetVersion(u.buildInfo.Version)
+
+	// Set resource attributes based on the first record
+	// (all records have the same account ID and region)
+	u.setResourceAttributes(resourceLogs.Resource().Attributes(), records[0])
+
+	for _, record := range records {
+		logRecord := scopeLogs.LogRecords().AppendEmpty()
+		u.setLogRecord(logRecord, record)
+	}
+
+	return logs
+}
+
+// checks if the event is related to EC2 instance operations
+func isEC2InstanceOperation(eventName string) bool {
+	ec2Operations := map[string]bool{
+		"StartInstances":     true,
+		"StopInstances":      true,
+		"TerminateInstances": true,
+		"RebootInstances":    true,
+		"RunInstances":       true,
+	}
+	return ec2Operations[eventName]
+}
+
+func extractInstanceIDs(requestParams map[string]any) []string {
+	var instanceIDs []string
+
+	if instancesSet, ok := requestParams["instancesSet"].(map[string]any); ok {
+		if items, ok := instancesSet["items"].([]any); ok {
+			for _, item := range items {
+				if itemMap, ok := item.(map[string]any); ok {
+					if instanceID, ok := itemMap["instanceId"].(string); ok {
+						instanceIDs = append(instanceIDs, instanceID)
+					}
+				}
+			}
+		}
+	}
+
+	return instanceIDs
+}
+
+func (u *CloudTrailLogsUnmarshaler) setResourceAttributes(attrs pcommon.Map, record CloudTrailRecord) {
+	attrs.PutStr(string(conventions.CloudProviderKey), conventions.CloudProviderAWS.Value.AsString())
+	attrs.PutStr(string(conventions.CloudRegionKey), record.AwsRegion)
+	attrs.PutStr(string(conventions.CloudAccountIDKey), record.RecipientAccountID)
+}
+
+func (u *CloudTrailLogsUnmarshaler) setLogRecord(logRecord plog.LogRecord, record CloudTrailRecord) {
+	if t, err := time.Parse(time.RFC3339, record.EventTime); err == nil {
```

Comment:
What if there is an error? Should it fail? Do you know what happens if we don't set a timestamp on a log record?

Reply:
Given this is part of the cloudtrail platform itself and not timestamps coming from user's code it is very unlikely this parsing would fail, but I changed the behavior to resemble s3-access-log for consistency and added a test.